packages/lodestar-beacon-state-transition/src/util/domain.ts

```diff
@@ -5,24 +5,37 @@ import {
   Epoch,
   Version,
   BeaconState,
+  Root,
 } from "@chainsafe/lodestar-types";
 import {IBeaconConfig} from "@chainsafe/lodestar-config";
 
-import {DomainType} from "../constants";
+import {DomainType, ZERO_HASH} from "../constants";
 import {intToBytes} from "@chainsafe/lodestar-utils";
 
 import {getCurrentEpoch} from "./epoch";
+import {computeForkDataRoot} from "./fork";
 
 /**
  * Return the domain for the [[domainType]] and [[forkVersion]].
  */
-export function computeDomain(config: IBeaconConfig, domainType: DomainType, forkVersion?: Version): Buffer {
+export function computeDomain(
+  config: IBeaconConfig,
+  domainType: DomainType,
+  forkVersion?: Version,
+  genesisValidatorRoot: Root = ZERO_HASH): Buffer {
   if (!forkVersion) {
     forkVersion = config.params.GENESIS_FORK_VERSION;
   }
+  const forkDataRoot = computeForkDataRoot(config, forkVersion, genesisValidatorRoot);
+  const forkDataRootForDomain = Buffer.alloc(28);
```

Comment:
The spec is just like this `fork_data_root[:28]`. Do we want to add `spice` or similar methods to our ssz `ArrayLike`?

Reply:
It would be nice to add, might be tricky to get the types right.
Until then we can do `(forkDataRoot.valueOf() as Uint8Array).slice(0,28)`

Reply:
thanks, changed it accordingly